### PR TITLE
[bitnami/cassandra] Making TLS existingSecrets more configurable

### LINKED ISSUE #28687: [bitnami/cassandra] Make TLS secret configuration more versatile to allow interaction with cert-manager/trust-manager
### Name and Version

bitnami/cassandra 11.3.12

### What is the problem this feature will solve?

We are trying to use the `bitnami/cassandra` chart to deploy a cassandra cluster with TLS enabled, using certificates/keystores created via [cert-manager](https://cert-manager.io/) and truststores created by trust-manager (part of the same project), which leads to issues due to the differing formats of the secrets created by cert-manager/trust-manager vs the secret expected by the chart.

The main differences are:

* chart expects a single secret for truststore and keystore, which doesn't allow usage of trust-manager which creates a separate secret for the truststore
* chart expects the keys `keystore` and `truststore`, but cert-manager and trust-manager create the keys `keystore.jks`/`keystore.p12` and `truststore.jks`/`truststore.p12` respectively (not configurable on either side)
* chart expects the keys `keystore-password` and `truststore-password` in the same secret, with non-configurable keys (this is configurable on cert-manager and trust-manager side though, so not a deal-breaker but still not flexible)

### What is the feature you are proposing to solve the problem?

Introduce the following values:

* `tls.secretKeys.truststorePassword` with default `truststore-password`
* `tls.secretKeys.keystorePassword` with default `keystore-password`
* `tls.secretKeys.keystore` with default `keystore`
* `tls.secretKeys.truststore` with default `truststore`
* `tls.existingTrustSecret` with default of empty string (signaling "none", meaning to use `tls.existingSecret` for both)

Then instead of hardcoded values for the secret keys, the respective values could be used while ensuring backwards compatibility. Special care has to be taken in case of `keystore` and `truststore` which are also used as filenames in init-containers and environment variables as the tls secret is mounted to `/certs` which could be replaced with a projected volume built from just the expected/required files from one or multiple secrets as configured.

### What alternatives have you considered?

Another alternative would be (at least for some users not requiring trust-manager) to request changes just in cert-manager to make those keys configurable. I personally think though that this is the more appropriate place.

Another alternative would be to build some kind of workaround that generates the expected secrets from the existing secrets generated by cert-manager before deploying the helm chart. This is however quite inconvenient.

### PR DESCRIPTION
**Pull request by tbalzer**
### Description of the change

Added values to allow for more configuration of the TLS settings, e.g. alternate secret keys, split secrets.

### Benefits

E.g. interoperability with cert-manager and trust-manager.

### Possible drawbacks

None really?

### Applicable issues

- fixes #28687 

### Additional information

TLS configuration is not currently covered by automated tests.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by tbalzer** [comment]
@carrodher I'd appreciate any help you can give me with getting the pipeline to succeed.

I have invited @bitnami-bot to the repository which could be accepted to resolve this.
As it's an organization owned fork I can't use the GitHub builtin-functionality to give access (as it's only for personally owned forks).

Unfortunately the log of the changelog job doesn't show the diff which I could apply and I seem to be unable to locally reproduce what this job generates.

File: bitnami/cassandra/values.yaml [hunk 1]
```diff
@@ -940,18 +941,28 @@ metrics:
...
+## @param tls.secretKeys.keystorePassword Key within `tls.passwordsSecret` that contains the keystore secret
+## @param tls.secretKeys.truststore Key within `tls.existingSecret` or `tls.existingTrustSecret` (if set) that contains the truststore file
+## @param tls.secretKeys.truststorePassword Key within `tls.passwordsSecret` that contains the truststore secret
 ##
 tls:
   internodeEncryption: none
   clientEncryption: false
   autoGenerated: false
   existingSecret: ""
+  existingTrustSecret: ""
```
Comment (migruiz4) [inline bitnami/cassandra/values.yaml:954]: For consistency, would it be possible to rename this as `existingTruststoreSecret`?

**Review by migruiz4 — COMMENTED** [review]
Thank you very much for your contribution @tbalzer! Could you please take a look at my comments?

File: bitnami/cassandra/templates/statefulset.yaml [hunk 2]
```diff
@@ -552,9 +552,26 @@ spec:
...
+                    - key: {{ .Values.tls.secretKeys.truststore }}
+                      path: truststore
+                      mode: 256
+              - secret:
+                  name: {{ include "cassandra.keystoreSecretName" . }}
+                  items:
+                    - key: {{ .Values.tls.secretKeys.keystore }}
+                      path: keystore
+                      mode: 256
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (tbalzer) [reply bitnami/cassandra/templates/statefulset.yaml:574]: The created secret from https://github.com/bitnami/charts/blob/main/bitnami/cassandra/templates/tls-secret.yaml still uses hardcoded keys, as those are then not in JKS format. This is then handled differently in the init-container.

I think it wouldn't make sense there to use a projected secret then?

File: bitnami/cassandra/values.yaml (hunk 1 above)
Comment (tbalzer) [reply bitnami/cassandra/values.yaml]: Yes, good idea. Done.

**Comment by tbalzer** [comment]
> Thank you very much for your contribution @tbalzer! Could you please take a look at my comments?

Thanks @migruiz4, I've incorporated all but one and left a reply the last. When you have time please take a look.

**Comment by tbalzer** [comment]
Hey @migruiz4 , this is getting stale right now. Anything I can do to move this forward?

**Comment by tbalzer** [comment]
> Sorry for the delay! Please note a new version was released in the meantime, you need to rebase from main and bump the chart version in the _Chart.yaml_ again

I've rebased and updated accordingly.

**Comment by tbalzer** [comment]
@carrodher @migruiz4 Can I get a review here?

File: bitnami/cassandra/templates/statefulset.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbalzer) [reply bitnami/cassandra/templates/statefulset.yaml:574]: @migruiz4 It's been a month now, any thoughts?

**Pull request closed without merging**